lib/sisimai/bite/email/sendmail.rb

```diff
@@ -0,0 +1,290 @@
+              # ... while talking to mta.example.org.:
+              # >>> DATA
+              # <<< 550 Unknown user recipient@example.jp
+              # 554 5.0.0 Service unavailable
+              # ...
+              # Reporting-MTA: dns; mx.example.jp
+              # Received-From-MTA: DNS; x1x2x3x4.dhcp.example.ne.jp
+              # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+              if cv = e.match(/\A[>]{3}[ ]+([A-Z]{4})[ ]?/)
+                # >>> DATA
+                commandtxt = cv[1]
+
+              elsif cv = e.match(/\A[<]{3}[ ]+(.+)\z/)
+                # <<< Response
+                esmtpreply = cv[1]
+
+              elsif cv = e.match(/\A[Rr]eporting-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Reporting-MTA: dns; mx.example.jp
+                next if connheader['rhost'].size > 0
+                connheader['rhost'] = cv[1].downcase
+                connvalues += 1
+
+              elsif cv = e.match(/\A[Rr]eceived-[Ff]rom-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Received-From-MTA: DNS; x1x2x3x4.dhcp.example.ne.jp
+                next if connheader['lhost']
+
+                # The value of "lhost" is optional
+                connheader['lhost'] = cv[1].downcase
+                connvalues += 1
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                connheader['date'] = cv[1]
+                connvalues += 1
+
+              else
+                # Detect SMTP session error or connection error
+                next if sessionerr
+                if e =~ Re1[:error]
+                  # ----- Transcript of session follows -----
+                  # ... while talking to mta.example.org.:
+                  sessionerr = true
+                  next
+                end
+
+                if cv = e.match(/\A[<](.+)[>][.]+ (.+)\z/)
+                  # <kijitora@example.co.jp>... Deferred: Name server: example.co.jp.: host name lookup failure
+                  anotherset['recipient'] = cv[1]
+                  anotherset['diagnosis'] = cv[2]
+
+                else
+                  # ----- Transcript of session follows -----
+                  # Message could not be delivered for too long
+                  # Message will be deleted from queue
+                  next if e =~ /\A[ \t]*[-]+/
+                  if cv = e.match(/\A[45]\d\d[ \t]([45][.]\d[.]\d)[ \t].+/)
+                    # 550 5.1.2 <kijitora@example.org>... Message
+                    #
+                    # DBI connect('dbname=...')
+                    # 554 5.3.0 unknown mailer error 255
+                    anotherset['status'] = cv[1]
+                    anotherset['diagnosis'] ||= ''
+                    anotherset['diagnosis']  += ' ' + e
+
+                  elsif e =~ /\A(?:Message|Warning:) /
+                    # Message could not be delivered for too long
```

Comment:
__[rubocop]__
* Metrics/BlockNesting: Avoid more than 4 levels of block nesting. [:ref](https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)

<!-- Comment by [SideCI](https://sideci.com) -->